File: solanaj/src/main/java/org/p2p/solanaj/kits/transaction/TransactionDetailsType.java
package org.p2p.solanaj.kits.transaction;

import androidx.annotation.NonNull;

public enum TransactionDetailsType {
    UNKNOWN("unknown"),
    SWAP("swap"),
    TRANSFER("transfer"),
    CREATE_ACCOUNT("create"),
    CLOSE_ACCOUNT("closeAccount");

    private String type;

    TransactionDetailsType(String type) {
        this.type = type;
    }

    @NonNull
    public String getType() {
        return type;
    }

}

File: solanaj/src/main/java/org/p2p/solanaj/kits/transaction/TransactionDetailsType.kt
package org.p2p.solanaj.kits.transaction

enum class TransactionDetailsType(val typeStr: String?) {
    SWAP("swap"),
    TRANSFER("transfer"),
    TRANSFER_CHECKED("transferChecked"),
    CREATE_ACCOUNT("create"),
    BURN_CHECKED("burnChecked"),
    CLOSE_ACCOUNT("closeAccount"),
    UNKNOWN("unknown");

    companion object {
        fun valueOf(typeStr: String?): TransactionDetailsType =
            values().firstOrNull { it.typeStr == typeStr } ?: UNKNOWN
    }
}
